django/utils/functional.py

```diff
@@ -132,6 +132,11 @@ def __cast(self):
             else:
                 return func(*self.__args, **self.__kw)
 
+        @property
+        def __class__(self):
+            # proxy __class__ to make isinstance(lazy_result, resultclass) work
+            return self.__cast().__class__
```

Comment:
This makes isinstance(ugettext_lazy(...), six.text_type) return True on both Python 2 and 3. It also breaks pickling of lazy() results on Python 2 (only). There's probably a way to fix that, but who knows what else pulling that thread might unravel.